packages/ERTP/src/interfaces.js

```diff
@@ -0,0 +1,47 @@
+import { assert, details, q, an } from '@agoric/assert';
+import { E } from '@agoric/eventual-send';
+
+const ERTPKind = {
+  ISSUER: 'issuer',
+  BRAND: 'brand',
+  PURSE: 'purse',
+  PAYMENT: 'payment',
+  MINT: 'mint',
+};
+
+const last = array => array[array.length - 1];
+
+const makeInterface = (allegedName, kind) => `Alleged: ${allegedName} ${kind}`;
+
+const makeAssertKindWhen = (getInterfaceOf, kind) => maybeRemotableP => {
+  return E.when(maybeRemotableP, maybeRemotable => {
+    const iface = getInterfaceOf(maybeRemotable);
+    const detailsMsg = details`${maybeRemotableP} must be ${q(
+      an(kind),
+    )} or a promise for ${q(an(kind))}`;
+    assert(iface, detailsMsg);
+    assert(kind === last(iface.split(' ')), detailsMsg);
+  });
+};
```

Comment:
Since the purpose of this is not ERTP specific, we may eventually want such supporting abstractions to be exported from a more reusable place. For example, the Zoe service, seats, and much else. To make it more reusable though, we wouldn't have a single enum of all interface names. Since I don't know how we'd make it nicely more reusable while staying within my current string kludge, I do not suggest doing it in this PR. Just noting the need.

Reply:
Sure, I think it makes sense to move the portions that are not ERTP specific out of ERTP when we start using them elsewhere.